dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,332 @@
+<template>
+  <div v-if="waitingForData" class="d-center-div text-center px-7 py-4">
+    <p class="font-medium text-xl">Loading P2P Data...</p>
+    <em class="pi pi-spinner pi-spin" aria-hidden="true" style="z-index: 20; color: #e67f3f" />
+  </div>
+  <div v-if="mapOfKpiKeysToDataObjectsArrays.size > 0 && !waitingForData">
+    <DataTable tableClass="onlyHeaders">
+      <Column bodyClass="headers-bg" headerStyle="width: 30vw;" headerClass="horizontal-headers-size" header="KPIs">
+      </Column>
+      <Column
+        v-for="reportingPeriodWithDataId of listOfDataSetReportingPeriods"
+        headerClass="horizontal-headers-size"
+        headerStyle="width: 30vw;"
+        :field="reportingPeriodWithDataId.dataId"
+        :header="reportingPeriodWithDataId.reportingPeriod"
+        :key="reportingPeriodWithDataId.dataId"
+      >
+      </Column>
+    </DataTable>
+    <div v-for="(arrayOfKpiDataObject, index) in mapOfKpiKeysToDataObjectsArrays" :key="index" class="d-table-style">
+      <div v-if="shouldCategoryBeRendered(arrayOfKpiDataObject[0])">
+        <div>
+          <div class="pt-2 pl-2 pb-2 w-full d-cursor-pointer border-bottom-table p-2" @click="toggleExpansion(index)">
+            <div v-if="!isExpanded(index)">
+              <span
+                :class="`p-badge badge-${colorOfCategory(arrayOfKpiDataObject[0])}`"
+                :data-test="arrayOfKpiDataObject[0]"
+                >{{ arrayOfKpiDataObject[0].toUpperCase() }}</span
+              >
+              <button class="pt-1 pr-3 d-cursor-pointer d-chevron-style">
+                <span class="pr-1 pt-1 pi pi-chevron-right d-chevron-font"></span>
+              </button>
+            </div>
+            <div v-if="isExpanded(index)">
+              <span :class="`p-badge badge-${colorOfCategory(arrayOfKpiDataObject[0])}`">{{
+                arrayOfKpiDataObject[0].toUpperCase()
+              }}</span>
+              <button class="pt-2 pr-3 d-cursor-pointer d-chevron-style">
+                <span class="pr-1 pi pi-chevron-down d-chevron-font"></span>
+              </button>
+            </div>
+          </div>
+        </div>
+        <div v-show="isExpanded(index)">
+          <DisplayFrameworkDataTable
+            :arrayOfKpiDataObjects="arrayOfKpiDataObject[1]"
+            :list-of-reporting-periods-with-data-id="listOfDataSetReportingPeriods"
+            headerInputStyle="display: none;"
```

Comment:
How about making this rather a boolean which decides internally to use `display: none` or `width: 30vw` as header style. I don't like that we must know the exact number `30` in LksgPanel.vue for this.

Reply:
We did it that way, because the assumption was that we will also switch lksg to this new layout and we did not want to invest more time into this.

Reply:
Good point with Lksg. Will be resolved.